packages/dcos-integration-test/extra/test_dcos_diagnostics.py

```diff
@@ -214,25 +213,25 @@ def test_systemd_units_health(dcos_api_session):
         raise AssertionError('\n'.join(unhealthy_output))
 
 
-def test_dcos_diagnostics_units_unit(dcos_api_session):
+def test_dcos_diagnostics_units_unit(dcos_api_session: Any) -> None:
     """
     test a unit response in a right format, endpoint: /system/health/v1/units/<unit>
     """
     for master in dcos_api_session.masters:
         units_response = check_json(dcos_api_session.health.get('/units', node=master))
-        pulled_units = list(map(lambda unit: unit['id'], units_response['units']))
+        pulled_units = list(map(lambda unit: unit['id'], units_response['units']))  # type: ignore
         for unit in pulled_units:
             unit_response = check_json(dcos_api_session.health.get('/units/{}'.format(unit), node=master))
             validate_units([unit_response])
 
 
 @retrying.retry(wait_fixed=2000, stop_max_delay=LATENCY * 1000)
-def test_dcos_diagnostics_units_unit_nodes(dcos_api_session):
+def test_dcos_diagnostics_units_unit_nodes(dcos_api_session: Any) -> None:
     """
     test a list of nodes for a specific unit, endpoint /system/health/v1/units/<unit>/nodes
     """
 
-    def get_nodes_from_response(response) -> List[str]:
+    def get_nodes_from_response(response: Any) -> list:
```

Comment:
We should keep this as `List[str]`. If it is triggering an error, have we found a bug in the test, or do we need more mypy annotations?

Reply:
In a case List[str] will be used here mypy will raise next errors:
```
packages/dcos-integration-test/extra/test_dcos_diagnostics.py:236: error: No overload variant of "__getitem__" of "list" matches argument type "str"
packages/dcos-integration-test/extra/test_dcos_diagnostics.py:236: note: Possible overload variants:
packages/dcos-integration-test/extra/test_dcos_diagnostics.py:236: note:     def __getitem__(self, int) -> str
packages/dcos-integration-test/extra/test_dcos_diagnostics.py:236: note:     def __getitem__(self, slice) -> List[str]
```

Reply:
I have play with this implementation, only example below are working:
```
    def get_nodes_from_response(response: Any) -> List[Optional[str]]:
        assert 'nodes' in response, 'response must have field `nodes`. Got {}'.format(response)
        nodes_ip_map = make_nodes_ip_map(dcos_api_session)  # type: Dict[str, str]
        nodes = []  # type: List[Optional[str]]
        for node in response['nodes']:
            assert 'host_ip' in node, 'node response must have `host_ip` field. Got {}'.format(node)
            assert node['host_ip'] in nodes_ip_map, 'nodes_ip_map must have node {}. Got {}'.format(
                node['host_ip'], nodes_ip_map)
            nodes.append(nodes_ip_map.get(node['host_ip']))
        return nodes
```
In a case I use only replacement like this:
```
    def get_nodes_from_response(response: Any) -> List[str]:
        assert 'nodes' in response, 'response must have field `nodes`. Got {}'.format(response)
        nodes_ip_map = make_nodes_ip_map(dcos_api_session)
        nodes = []  # type: List[Optional[str]]
```
I got
```
packages/dcos-integration-test/extra/test_dcos_diagnostics.py:242: error: Incompatible return value type (got "List[Optional[str]]", expected "List[str]")
```
I think 'List[Optional[str]]' is a option because we do not have `str` type check in this function.